game-core/src/test/java/games/strategy/triplea/odds/calculator/RunCountDistributorTest.java

```diff
@@ -0,0 +1,74 @@
+package games.strategy.triplea.odds.calculator;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.util.stream.IntStream;
+import org.junit.jupiter.api.Test;
+
+public class RunCountDistributorTest {
+
+  @Test
+  void verifyEvenDistributionWhenCleanlyDividable() {
+    final var runCountDistributor = new RunCountDistributor(20, 4);
+
+    assertThat(runCountDistributor.nextRunCount(), is(equalTo(5)));
```

Comment:
Minor suggestion: Might be good to have a display name to say we are spawning 4 threads and each should have 5. It's not immediately clear why there is repetition.